polarCoordinateTraversalTest.m

```diff
@@ -278,15 +278,211 @@ function testRayGoesAlongThetaBoundary(testCase)
+        expected_rVoxels     = [1,1];
+        expected_thetaVoxels = [2,3];
+        
+        verifyEqual(testCase, rVoxels, expected_rVoxels);
+        verifyEqual(testCase, thetaVoxels, expected_thetaVoxels);
+end
+
 % - Ray with positive x- and negative y- directions.
+function testRayWithPositiveXNegativeY(testCase)
+    min_bound = [0.0, 0.0];
+    max_bound = [30.0, 30.0];
+    ray_origin = [10.0, 25.0];
+    ray_direction = [2.0, -2.0];
+    circle_center = [15.0, 15.0];
+    circle_max_radius = 10.0;
+    num_radial_sections = 3;
+    num_angular_sections = 4;
+    t_begin = 0.0;
+    t_end = 20.0;
+    verbose = false;
+    
+    [rVoxels, thetaVoxels] = polarCoordinateTraversal(min_bound, max_bound, ray_origin, ray_direction, ...
+        circle_center, circle_max_radius, num_radial_sections, num_angular_sections, t_begin, t_end, verbose);
+        expected_rVoxels     = [1,2,2,2,1];
+        expected_thetaVoxels = [1,1,0,3,3];
+        
+        verifyEqual(testCase, rVoxels, expected_rVoxels);
+        verifyEqual(testCase, thetaVoxels, expected_thetaVoxels);
+end
+
 % - Ray with negative x- and positive y- directions.
+function testRayWithNegativeXPositiveY(testCase)
+    min_bound = [0.0, 0.0];
+    max_bound = [30.0, 30.0];
+    ray_origin = [10.0, 25.0];
+    ray_direction = [2.0, -2.0];
+    circle_center = [15.0, 15.0];
+    circle_max_radius = 10.0;
+    num_radial_sections = 3;
+    num_angular_sections = 4;
+    t_begin = 0.0;
+    t_end = 20.0;
+    verbose = false;
+    
+    [rVoxels, thetaVoxels] = polarCoordinateTraversal(min_bound, max_bound, ray_origin, ray_direction, ...
+        circle_center, circle_max_radius, num_radial_sections, num_angular_sections, t_begin, t_end, verbose);
+        expected_rVoxels     = [1,1];
+        expected_thetaVoxels = [3,2];
+        
+        verifyEqual(testCase, rVoxels, expected_rVoxels);
+        verifyEqual(testCase, thetaVoxels, expected_thetaVoxels);
+end
+
 % - Ray that has a t_begin time different than 0.0
+function testRayBeginTimeDiffThan0(testCase)
+    min_bound = [0.0, 0.0];
+    max_bound = [30.0, 30.0];
+    ray_origin = [10.0, 25.0];
+    ray_direction = [2.0, -2.0];
+    circle_center = [15.0, 15.0];
+    circle_max_radius = 10.0;
+    num_radial_sections = 3;
+    num_angular_sections = 4;
+    t_begin = 10.0;
+    t_end = 20.0;
+    verbose = false;
+    
+    [rVoxels, thetaVoxels] = polarCoordinateTraversal(min_bound, max_bound, ray_origin, ray_direction, ...
+        circle_center, circle_max_radius, num_radial_sections, num_angular_sections, t_begin, t_end, verbose);
+        expected_rVoxels     = [2,2,1];
+        expected_thetaVoxels = [1,2,2];
+        
+        verifyEqual(testCase, rVoxels, expected_rVoxels);
+        verifyEqual(testCase, thetaVoxels, expected_thetaVoxels);
+end
+
 % - Ray that has an odd number of angular sections.
+function testRayWithOddNumAngularSections(testCase):
```

Comment:
This doesn't have an odd number of angular sections, and it looks like it doesn't traverse any theta voxels

Reply:
Actually I'm a little bit confused. When you say there's an odd number of angular sections, what do you mean? Like the ray intersect angular sections with odd number of times?

Reply:
the ```num_angular_sections``` is not even, i.e. ```num_angular_sections = 5;```